### PR TITLE
Feat/rpc module

### PR DESCRIPTION
**Pull request by gnkz**
Added `rpc` module using the `vm.rpc` cheatcode

### CODE REVIEW

File: src/_internal/Rpc.sol [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import {vulcan} from "./Vulcan.sol";
+import {forksUnsafe, Fork} from "./Forks.sol";
+import {JsonObject} from "./Json.sol";
+
+library rpc {
+    /// @dev Calls an JSON-RPC method on a specific RPC endpoint. If there was a previous active fork it will return back to that one once the method is called.
+    /// @param url The url of the RPC endpoint to use
+    /// @param method The JSON-RPC method to call
+    /// @param params The method params as a JSON string
+    function call(string memory url, string memory method, string memory params) internal returns (bytes memory data) {
```
Comment (vdrg) [inline src/_internal/Rpc.sol:13]: maybe it is better to implement all the logic in the `call` function that receives a JsonObject, that way the params are validated?

File: src/_internal/Rpc.sol [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        return result;
+    }
+
+    /// @dev Calls an JSON-RPC method on a specific RPC endpoint. If there was a previous active
+    /// fork it will return back to that one once the method is called.
+    /// @param url The url of the RPC endpoint to use
+    /// @param method The JSON-RPC method to call
+    /// @param params The method params as a JsonObject
+    function call(string memory url, string memory method, JsonObject memory params)
```
Comment (vdrg) [inline src/_internal/Rpc.sol:38]: maybe it should be urlOrName in the future (not necessary now).

File: test/examples/rpc/UtilsExample01.t.sol [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MIT
```
Comment (vdrg) [inline test/examples/rpc/UtilsExample01.t.sol:1]: The filename is UtilsExample

File: test/examples/rpc/UtilsExample01.t.sol [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+        bytes memory data = rpc.call(method, params);
+
+        uint8 chainId;
+
+        assembly {
+            chainId := mload(add(data, 0x01))
+        }
+
+        expect(chainId).toEqual(1);
```
Comment (vdrg) [inline test/examples/rpc/UtilsExample01.t.sol:25]: maybe do block.chainId

File: src/_internal/Rpc.sol (hunk 1 above)
Comment (gnkz) [reply src/_internal/Rpc.sol]: That's a good idea. Should be solved now

File: src/_internal/Rpc.sol (hunk 2 above)
Comment (gnkz) [reply src/_internal/Rpc.sol]: yup, updated

File: test/examples/rpc/UtilsExample01.t.sol (hunk 4 above)
Comment (gnkz) [reply test/examples/rpc/UtilsExample01.t.sol]: My bad. Fixed

**Pull request merged**